pkg/cloud/vsphere/services/metadata/metadata_test.go

```diff
@@ -0,0 +1,216 @@
+					MachineSpec: v1alpha1.VsphereMachineSpec{
+						Network: v1alpha1.NetworkSpec{
+							Devices: []v1alpha1.NetworkDeviceSpec{
+								{
+									NetworkName: "network1",
+									MACAddr:     "00:00:00:00:00",
+									DHCP6:       true,
+									IPAddrs:     []string{"192.168.4.21"},
+									Gateway4:    "192.168.4.1",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "static4+dhcp6+static-routes",
+			ctx: &context.MachineContext{
+				Machine: &clusterv1.Machine{},
+				MachineConfig: &v1alpha1.VsphereMachineProviderConfig{
+					MachineSpec: v1alpha1.VsphereMachineSpec{
+						Network: v1alpha1.NetworkSpec{
+							Devices: []v1alpha1.NetworkDeviceSpec{
+								{
+									NetworkName: "network1",
+									MACAddr:     "00:00:00:00:00",
+									DHCP6:       true,
+									IPAddrs:     []string{"192.168.4.21"},
+									Gateway4:    "192.168.4.1",
+								},
+							},
+							Routes: []v1alpha1.NetworkRouteSpec{
+								{
+									To:     "192.168.5.1/24",
+									Via:    "192.168.4.254",
+									Metric: 3,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "2nets",
+			ctx: &context.MachineContext{
+				Machine: &clusterv1.Machine{},
+				MachineConfig: &v1alpha1.VsphereMachineProviderConfig{
+					MachineSpec: v1alpha1.VsphereMachineSpec{
+						Network: v1alpha1.NetworkSpec{
+							Devices: []v1alpha1.NetworkDeviceSpec{
+								{
+									NetworkName: "network1",
+									MACAddr:     "00:00:00:00:00",
+									DHCP4:       true,
+									Routes: []v1alpha1.NetworkRouteSpec{
+										{
+											To:     "192.168.5.1/24",
+											Via:    "192.168.4.254",
+											Metric: 3,
+										},
+									},
+								},
+								{
+									NetworkName: "network12",
+									MACAddr:     "00:00:00:00:01",
+									DHCP6:       true,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "2nets-static+dhcp",
+			ctx: &context.MachineContext{
+				Machine: &clusterv1.Machine{},
+				MachineConfig: &v1alpha1.VsphereMachineProviderConfig{
+					MachineSpec: v1alpha1.VsphereMachineSpec{
+						Network: v1alpha1.NetworkSpec{
+							Devices: []v1alpha1.NetworkDeviceSpec{
+								{
+									NetworkName:   "network1",
+									MACAddr:       "00:00:00:00:00",
+									IPAddrs:       []string{"192.168.4.21"},
+									Gateway4:      "192.168.4.1",
+									Nameservers:   []string{"1.1.1.1"},
+									SearchDomains: []string{"vmware.ci"},
+								},
+								{
+									NetworkName:   "network12",
+									MACAddr:       "00:00:00:00:01",
+									DHCP6:         true,
+									SearchDomains: []string{"vmware6.ci"},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.ctx.Machine.Name = tc.name
+			actVal, err := metadata.New(tc.ctx)
+			if err != nil {
+				t.Fatal(err)
+			}
+			buf, err := base64.StdEncoding.DecodeString(actVal)
```

Comment:
nit: validate against actual string values?

Reply:
Unless I have authoritative, expected data, any generated data is going to be based on previous generated data. For now I just wanted to make sure *looked* okay.